### PR TITLE
Add script and improve integration tests

### PR DESCRIPTION
**Pull request by Thumimku**
In order to support Nashorn deprecation we are gonna use openjdk Nashorn in integration tests.

We will improve the test groups also.

We are going to add new PR Builder for JDK 11 and update dependency updater to JDK 11,

Issue: https://github.com/wso2/product-is/issues/13463

### CODE REVIEW

File: modules/distribution/bin/dependencydownloader.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
```
Comment (piraveena) [inline modules/distribution/bin/dependencydownloader.sh:1]: can we rename this script name more meaningful?
Eg: adaptivescript.sh, nashron.sh

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    private boolean waitForMessage(ServerLogReader inputStreamHandler,
+                                  String message) {
+        long time = System.currentTimeMillis() + 60 * 1000;
+        while (System.currentTimeMillis() < time) {
+            if (inputStreamHandler.getOutput().contains(message)) {
+                return true;
+            }
+        }
+        return false;
+    }
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java:98]: Add a new line after this method

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptTemporaryClaimPersistenceTestCase.java [hunk 3]
```diff
@@ -101,6 +105,9 @@ public class AdaptiveScriptTemporaryClaimPersistenceTestCase extends AbstractAda
     public void testInit() throws Exception {
 
         super.init();
+//        runAdaptiveAuthenticationDependencyScript();
+//         //Re-initiating after the restart.
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptTemporaryClaimPersistenceTestCase.java:109]: remove these commented lines

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java:2]: should be 2022

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    private void runAdaptiveAuthenticationDependencyScript() {
+
+        ServerLogReader inputStreamHandler;
+        ServerLogReader errorStreamHandler;
+        String scriptFolder = (System.getProperty("carbon.home"))
+                + File.separator + "bin" + File.separator;
+        Process tempProcess = null;
+        File shFile = new File(scriptFolder);
```
Comment (piraveena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java:54]: rename this variable

File: modules/distribution/bin/dependencydownloader.sh [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+if compgen -G "$CARBON_HOME/lib/nashorn-core-*.jar" > /dev/null; then
+    location=$(find "$CARBON_HOME/lib/" ~+ -type f -name "nashorn-core-*.jar" | head -1)
+    full_artifact_name=$(basename ${location})
+    artifact_name=$(echo "$full_artifact_name" | awk -F'-' '{print $3}')
+    LOCAL_NASHORN_VERSION=$(echo "$artifact_name" | awk -F'.' '{print $1 "." $2}')
+    if [ "$NASHORN_VERSION" = "$LOCAL_NASHORN_VERSION" ]; then
+    	echo "Updated nashorn exists. No need to download."
+    else 
+    	echo "Updated nashorn not found. Updating OpenJDK Nashorn."
```
Comment (senthalan) [inline modules/distribution/bin/dependencydownloader.sh:60]: Better to reward these messaging.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        Runtime rt = Runtime.getRuntime();
+
+        try {
+            if (System.getProperty("os.name").toLowerCase().contains("windows")) {
+                log.warn("Operating System is Windows, skipping execution");
+            } else {
+                log.info("Operating system is not windows. Executing shell script");
+                tempProcess = rt.getRuntime().exec(
+                        new String[] { "/bin/bash", "dependencydownloader.sh" }, null, shFile);
```
Comment (senthalan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java:63]: better to come up with a different name? it is not simply a dependency downloader right ? are we planning to extend this to download new dependencies?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        ServerLogReader errorStreamHandler;
+        String scriptFolder = (System.getProperty("carbon.home"))
+                + File.separator + "bin" + File.separator;
+        Process tempProcess = null;
+        File shFile = new File(scriptFolder);
+        Runtime rt = Runtime.getRuntime();
+
+        try {
+            if (System.getProperty("os.name").toLowerCase().contains("windows")) {
+                log.warn("Operating System is Windows, skipping execution");
```
Comment (senthalan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java:59]: we need to have the bat file. Are we planning to it this in the next iteration?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptTemporaryClaimPersistenceTestCase.java [hunk 9]
```diff
@@ -57,12 +57,15 @@
...
 import org.wso2.carbon.identity.application.common.model.xsd.ServiceProvider;
 import org.wso2.carbon.integration.common.admin.client.AuthenticatorClient;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
 import org.wso2.identity.integration.common.clients.application.mgt.ApplicationManagementServiceClient;
 import org.wso2.identity.integration.common.clients.oauth.OauthAdminClient;
 import org.wso2.identity.integration.test.util.Utils;
 import org.wso2.identity.integration.test.utils.DataExtractUtil;
 import org.wso2.identity.integration.test.utils.OAuth2Constant;
 
+import javax.xml.xpath.XPathExpressionException;
```
Comment (senthalan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptTemporaryClaimPersistenceTestCase.java:67]: wrong package order

File: modules/distribution/bin/dependencydownloader.sh (hunk 1 above)
Comment (IsuraD) [reply modules/distribution/bin/dependencydownloader.sh:1]: enableadaptive.sh

File: modules/distribution/bin/dependencydownloader.sh [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+    	rm $location
+    	wget -P $CARBON_HOME/lib https://repo1.maven.org/maven2/org/openjdk/nashorn/nashorn-core/$NASHORN_VERSION/nashorn-core-$NASHORN_VERSION.jar
+    fi
+else
+   echo "OpenJDK Nashorn not found. Downloading OpenJDK Nashorn."
+   wget -P $CARBON_HOME/lib https://repo1.maven.org/maven2/org/openjdk/nashorn/nashorn-core/$NASHORN_VERSION/nashorn-core-$NASHORN_VERSION.jar
```
Comment (IsuraD) [inline modules/distribution/bin/dependencydownloader.sh:66]: print a log after successfully running the script and copy the jar.

File: modules/distribution/bin/dependencydownloader.sh [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+if compgen -G "$CARBON_HOME/lib/nashorn-core-*.jar" > /dev/null; then
+    location=$(find "$CARBON_HOME/lib/" ~+ -type f -name "nashorn-core-*.jar" | head -1)
+    full_artifact_name=$(basename ${location})
+    artifact_name=$(echo "$full_artifact_name" | awk -F'-' '{print $3}')
+    LOCAL_NASHORN_VERSION=$(echo "$artifact_name" | awk -F'.' '{print $1 "." $2}')
+    if [ "$NASHORN_VERSION" = "$LOCAL_NASHORN_VERSION" ]; then
+    	echo "Updated nashorn exists. No need to download."
+    else 
+    	echo "Updated nashorn not found. Updating OpenJDK Nashorn."
+    	rm $location
```
Comment (IsuraD) [inline modules/distribution/bin/dependencydownloader.sh:61]: print a log after successfully running the script and copy the jar. print the removed jar version + added jar version

File: modules/distribution/bin/dependencydownloader.sh (hunk 10 above)
Comment (IsuraD) [reply modules/distribution/bin/dependencydownloader.sh:66]: Do the same thing in all the places

File: modules/distribution/bin/dependencydownloader.sh [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    	echo "Updated asm util not found. Updating asm."
+    	rm $location
+        wget -P $CARBON_HOME/components/lib https://repo1.maven.org/maven2/org/ow2/asm/asm-util/$ASM_VERSION/asm-util-$ASM_VERSION.jar
+    fi
+else
+   echo "asm util not found. Downloading asm."
+   wget -P $CARBON_HOME/components/lib https://repo1.maven.org/maven2/org/ow2/asm/asm-util/$ASM_VERSION/asm-util-$ASM_VERSION.jar
+fi
+
+echo "Updating Adaptive Authentication Dependencies finished."
```
Comment (piraveena) [inline modules/distribution/bin/dependencydownloader.sh:86]: Need to add below 
```
<file>
            <source>src/bin/dependencydownloader.sh</source>
            <outputDirectory>wso2is-${pom.version}/bin/</outputDirectory>
            <fileMode>744</fileMode>
        </file>
``` 
in the assembly/bin.xml file

Comment (piraveena) [inline modules/distribution/bin/dependencydownloader.sh:86]: add new line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java (hunk 8 above)
Comment (Thumimku) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AdaptiveScriptInitializerTestCase.java:59]: We will do that in next iteration.

**Review by piraveena — APPROVED** [review]

**Comment by piraveena** [comment]
@Thumimku can you link the git issue here?

**Comment by Thumimku** [comment]
Issue: https://github.com/wso2/product-is/issues/13463

File: modules/integration/tests-integration/tests-backend/src/test/resources/enableadaptive.sh [hunk 13]
```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
```
Comment (senthalan) [inline modules/integration/tests-integration/tests-backend/src/test/resources/enableadaptive.sh:1]: any particular reason to duplicate this script file

**Review by senthalan — APPROVED** [review]

**Pull request merged**